### DISCUSSION TITLE
Request: Write and run custom tools with Node/JavaScript

### DISCUSSION THREAD

**Discussion by CyrisXD** [open-webui/open-webui#5285] [Ideas]
**Describe the solution you'd like**
I would love to be able to use the tool editor to write and run custom tools built with Node/JavaScript instead of just Python.

**Describe alternatives you've considered**
Running a seperate NodeJS server hosted somewhere, but it's overkill.

**Comment by devbyray** [comment]
I would love to write a tool using JavaScript, similar to what is currently done in Python. Is that possible?

**Comment by valdirdd** [comment]
I did that using request library in python. I Just call an api that I have running in nodejs.
But would be great to have tha hability to run JS or PY

**Comment by eatmyvenom** [comment]
so is this on the table at all?
